src/Quasar/QuasarF/Interpreter/Internal.purs

```diff
@@ -149,17 +149,33 @@ mkRequest
   . (String → Either Error a)
   → AXF.AffjaxF RequestContent String
   → Free (Coproduct l AXFP) (Either QError a)
-mkRequest f = map (handleResult f) <<< liftF <<< right
+mkRequest f a =
+  map (map snd) $ mkRequest' f a
+
+mkRequest'
+  ∷ ∀ a l
+  . (String → Either Error a)
+  → AXF.AffjaxF RequestContent String
+  → Free (Coproduct l AXFP) (Either QError (Tuple (AX.AffjaxResponse String) a))
+mkRequest' f = map (handleResult f) <<< liftF <<< right
+
+withExpired ∷ ∀ a b. Tuple (AX.AffjaxResponse a) b → { content ∷ b, expired ∷ Boolean }
+withExpired (Tuple { headers: hs } content) =
+  { content
+  -- , expired: doSomethingWithHeaders $ Debug.spy hs
+  , expired: doSomethingWithHeaders hs
+  }
+  where
+  doSomethingWithHeaders headers = false
 
 handleResult
   ∷ ∀ a
   . (String → Either Error a)
   → Either Error (AX.AffjaxResponse String)
-  → Either QError a
-handleResult f =
-  case _ of
-    Right { status: StatusCode code, response, headers }
-      | code >= 200 && code < 300 → lmap Error (f response)
+  → Either QError (Tuple (AX.AffjaxResponse String) a)
+handleResult f = case _ of
+    Right resp@{ status: StatusCode code, response, headers }
+      | code >= 200 && code < 300 → map (Tuple resp) (lmap Error (f response))
```

Comment:
`map` is same as `rmap` and when you do mapl and mapr it's same as just doing one `bimap` so this could be written like `bimap Error (Tuple resp) (f response)`